File: src/main/java/com/github/joelws/release/tracker/conversion/ArtifactAdapter.java
package com.github.joelws.release.tracker.conversion;

import com.github.joelws.release.tracker.dto.artifact.ArtifactDto;
import com.github.joelws.release.tracker.entity.artifact.Artifact;
import com.github.joelws.release.tracker.entity.artifact.ArtifactPK;
import com.github.joelws.release.tracker.interfaces.Adapter;

import org.apache.log4j.Logger;

public class ArtifactAdapter implements Adapter<ArtifactDto, Artifact> {
    private static final Logger LOGGER = Logger.getLogger(ArtifactAdapter.class);

    @Override
    public Artifact adapt(ArtifactDto artifactDto) {
        LOGGER.info("Adapt - in: " + artifactDto.getClass());

        Artifact out = new Artifact();
        ArtifactPK artifactPK = new ArtifactPK();

        artifactPK.setGroupId(artifactDto.getGroupId());
        artifactPK.setArtifactId(artifactDto.getArtifactId());
        artifactPK.setVersion(artifactDto.getVersion());
        out.setId(artifactPK);

        LOGGER.info("Adapt - out: " + out.getClass());
        return out;
    }
}

File: src/main/kotlin/com/joelws/release/tracker/conversion/ArtifactAdapter.kt
package com.joelws.release.tracker.conversion

import com.joelws.release.tracker.entity.artifact.Artifact
import com.joelws.release.tracker.interfaces.Adapter
import com.joelws.release.tracker.model.artifact.ArtifactModel

open class ArtifactAdapter : Adapter<Artifact, ArtifactModel> {


    override fun adapt(incoming: Artifact): ArtifactModel {

        val out = ArtifactModel(groupId = incoming.id?.groupId!!,
                artifactId = incoming.id?.artifactId!!,
                version = incoming.id?.version!!)
        return out
    }
}
